caddyhttp/proxy/upstream_test.go

```diff
@@ -225,4 +226,39 @@ func TestParseBlock(t *testing.T) {
 			}
 		}
 	}
+
+	// tests for insecure skip verify
+	//re, _ := http.NewRequest("GET", "/", nil)
+	isv_re := []struct {
+		config string
+		flag   bool
+	}{
+		// Test #1: without flag
+		{"proxy / localhost:8080", false},
+
+		// Test #2: with flag
+		{"proxy / localhost:8080 {\n insecure_skip_verify \n}", true},
+	}
+
+	for i, test := range isv_re {
+		upstreams, err := NewStaticUpstreams(caddyfile.NewDispenser("Testfile", strings.NewReader(test.config)))
+		if err != nil {
+			t.Error("Expected no error. Got:", err.Error())
+		}
+		for _, upstream := range upstreams {
+			staticUpstream, ok := upstream.(*staticUpstream)
+			if !ok {
+				t.Error("Type mismatch")
+			}
+			transport, ok := staticUpstream.HealthCheck.Client.Transport.(*http.Transport)
```

Comment:
Yup, the client doesn't get created unless health_check is defined: https://github.com/mholt/caddy/blob/74664fe0467cc99f84fa64aa7175267b016366b0/caddyhttp/proxy/upstream.go#L109